pkg/scheduler/framework/runtime/framework.go

```diff
@@ -102,38 +115,70 @@ func (frw *frameworkImpl) RunScorePlugins(ctx context.Context, placement *policy
 	defer func() {
 		metrics.FrameworkExtensionPointDuration.WithLabelValues(score, result.Code().String()).Observe(utilmetrics.DurationInSeconds(startTime))
 	}()
-	pluginToClusterScores := make(framework.PluginToClusterScores, len(frw.filterPlugins))
-	for _, p := range frw.scorePlugins {
-		var scoreList framework.ClusterScoreList
-		for _, cluster := range clusters {
+	pluginToClusterScores := make(framework.PluginToClusterScores, len(frw.scorePlugins))
+	for _, pl := range frw.scorePlugins {
+		pluginToClusterScores[pl.Name()] = make(framework.ClusterScoreList, len(clusters))
+	}
+	ctx, cancel := context.WithCancel(ctx)
+	errCh := parallelize.NewErrorChannel()
+
+	// Run Score method for each cluster in parallel
+	frw.parallelizer.Until(ctx, len(clusters), func(index int) {
+		for _, p := range frw.scorePlugins {
+			cluster := clusters[index]
 			s, res := frw.runScorePlugin(ctx, p, placement, spec, cluster)
 			if !res.IsSuccess() {
-				return nil, framework.AsResult(fmt.Errorf("plugin %q failed with: %w", p.Name(), res.AsError()))
+				err := fmt.Errorf("plugin %q failed with: %w", p.Name(), res.AsError())
+				errCh.SendErrorWithCancel(err, cancel)
```

Comment:
errCh imports new code, how about just using a channel, like:
```go
	errCh := make(chan error, 1)
	...
	errCh <- err
	...
	if len(errCh) > 0 {
		xxx
	}
```

Reply:
In fact, Kubernetes is also implemented in this way.